src/contrib/cluster/Akka.Cluster.Sharding/PersistentShard.cs

```diff
@@ -69,7 +69,46 @@ public class PersistentShard : Shard
 
         protected override bool ReceiveCommand(object message)
         {
-            return HandleCommand(message);
+            switch (message)
+            {
+                case SaveSnapshotSuccess m:
+                    Log.Debug("PersistentShard snapshot saved successfully");
+                    /*
+                    * delete old events but keep the latest around because
+                    *
+                    * it's not safe to delete all events immediate because snapshots are typically stored with a weaker consistency
+                    * level which means that a replay might "see" the deleted events before it sees the stored snapshot,
+                    * i.e. it will use an older snapshot and then not replay the full sequence of events
+                    *
+                    * for debugging if something goes wrong in production it's very useful to be able to inspect the events
+                    */
+                    var deleteToSequenceNr = m.Metadata.SequenceNr - Settings.TunningParameters.KeepNrOfBatches * Settings.TunningParameters.SnapshotAfter;
+                    if (deleteToSequenceNr > 0)
+                    {
+                        DeleteMessages(deleteToSequenceNr);
+                    }
+                    break;
+                case SaveSnapshotFailure m:
+                    Log.Warning("PersistentShard snapshot failure: {}", m.Cause.Message);
```

Comment:
In .NET logging we must provide index for arguments. So `{}` itself will fail, you need to use `{0}`, `{1}` and so on.